_posts/yyyy-mm-dd-ansible-czy-da-sie-uporzadkowac-chaos.md

```diff
@@ -0,0 +1,301 @@
+mongodb_host: eximee-mongo
+mongo_db_auth: MONGODB-CR
+...
+```
+W pliku inventory znajdują się zatem dane dotyczące hostów, używanych portów czy dane autentykacyjne. W inventory warto umieścić również takie zmienne, które są częściej wykorzystywane w playbookach lub zależne są np. od serwerów, na którym instalujemy aplikację. Nie powinno się bowiem modyfikować playbooków przy okazji każdej instalacji, a raczej utrzymywać dla danego serwera odpowiednią konfigurację. W przypadku platformy Eximee takimi zmiennymi są np. adresy URL różnych aplikacji. Warto też zwrócić uwagę, że adresy w powyższym przykładzie nie są zdefiniowane „na sztywno”. Zawierają one odwołania do kilku innych zmiennych tworząc nową wartość, na co również pozwala Ansible.
+
+Wiedząc już czym jest moduł oraz gdzie należy umieścić konfigurację, potrzebna jest jeszcze wiedza, jak definiować pewien stan systemu, który chcemy osiągnąć. Do tego właśnie służą **playbooki**. Definicja takiego stanu dzieli się na taski. Zwiększa to nie tylko czytelność kodu playbooka, ale też oddziela od siebie niezależne etapy instalacji. 
+```
+---
+- name: Create installation_tmp dir
+  hosts: localhost
+  become: no
+  connection: local
+  tasks:
+    - file:
+        path: "{{ eximee_platform_dest }}/installation_tmp"
+        state: directory
+
+- import_playbook: check-settings.yml
+- import_playbook: eximee-common.yml
+- import_playbook: dmzzew.yml
+- import_playbook: dmz.yml
+- import_playbook: formstore.yml
+
+- name: Remove installation_tmp dir
+  hosts: localhost
+  become: no
+  gather_facts: false
+  connection: local
+  tasks:
+    - file:
+        path: "{{ eximee_platform_dest }}/installation_tmp"
+        state: absent
+```
+W powyższym przykładzie widoczne są dwa proste zadania. Pierwsze, o nazwie Create installation_tmp dir tworzy tymczasowy katalog instalacyjny installation_tmp w lokalizacji, która zdefiniowana zostala pod zmienną eximee_platform_dest. Drugie (Remove installation_tmp dir) ten katalog uswuwa. Pomiędzy tymi zadaniami zaimportowane zostały odseparowane logicznie playbooki. Każdy z nich instaluje odrębną część platformy, a ich rozdzielenie jeszcze bardziej zwiększa czytelność kodu. Co ciekawe, również importowane playbooki zostały odpowiednio podzielone na **role**, które na podstawie określonej struktury plików, pozwalają na automatyczne ładowanie zmiennych czy tasków. Ponadto, grupowanie zadań w role pozwala na ich łatwe reużywanie.
+
+Jak uruchomić taki skrypt instalacyjny? Nic prostszego – wystarczy poniższa instrukcja: `ansible-playbook nazwa_playbooka.yml`
+
+Docelowo więc pierwotna struktura paczki wdrożeniowej, składającej się ze skryptów bashowych, domyślnych parametrów konfiguracyjnych i plików z platformą nabrała nowego charakteru. Od teraz na paczkę składało się wiele plików z odseperowanymi logicznie zadaniami, które nie musiały już być rozpakowywane na maszynie docelowej, a jedynie uruchomione z odpowiednim użytkownikiem z poziomu maszyny kontrolnej:
```

Comment:
i plików z platformą **(przecinek)** nabrała nowego charakteru.

Dziwnie też brzmi następne zdanie, bo jest "Od teraz", a potem używasz czasu przeszłego, może da się to jakoś przeredagować?

Reply:
poprawione